packages/chevrotain/test/parse/traits/perf_tracer_spec.ts

```diff
@@ -21,7 +21,7 @@ skipOnBrowser("Chevrotain's Init Performance Tracing", () => {
 
     const PlusTok = createToken({ name: "PlusTok" })
 
-    class TraceParser extends Parser {
+    class TraceParser extends CstParser {
```

Comment:
was this changed needed to avoid an error?
I am trying to understand if there is a breaking change here?

Reply:
If we don't change the parent type to `CstParser`, the compiler won't be able to decide the type of nodes. My understanding is that this is the desired behavior, and it is why you added the two `Parser` types. right?

Reply:
Adding the two Parser types was indeed meant for cases like this.
What I am worried about is avoiding a **breaking change** if possible.
Or understanding if this change forces a breaking change in the APIs and thus a new major version